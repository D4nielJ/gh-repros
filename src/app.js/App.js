import Search from "../features/home/search"

const example = process.env.REACT_APP_EXAMPLE;

const App = () => {
  return (
    <div className="App">
      <Search />
      <div>{example}</div>
    </div>
  );
}

export default App;
