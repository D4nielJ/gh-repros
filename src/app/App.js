import { Switch, Route, useLocation } from 'react-router-dom';
import Search from '../features/home/search';
import User from '../features/user/user';

const useQuery = () => new URLSearchParams(useLocation().search);

const App = () => {
  const query = useQuery();

  return (
    <div className="App">
      <Switch>
        <Route path="/user">
          <User id={query.get('id')} />
        </Route>
        <Route exact path="/">
          <Search />
        </Route>
      </Switch>
    </div>
  );
};

export default App;
