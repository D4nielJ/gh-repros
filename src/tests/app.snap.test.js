import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';

import App from '../app/App';

it('renders correctly the app', () => {
  const tree = renderer
    .create(
      <Router>
        <App />
      </Router>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
