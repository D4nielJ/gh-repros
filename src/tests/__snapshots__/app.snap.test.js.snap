// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the app 1`] = `
<div
  className="App"
>
  <section
    className="px-4 py-40 flex flex-col items-center bg-bh-black min-h-screen"
  >
    <h1
      className="font-black text-5xl italic tracking-tighter text-bh-lightBlue mb-8"
    >
      repros
    </h1>
    <form
      className="md:w-1/3 flex flex-col justify-center items-center bg-bh-lightBlue px-6 py-12 rounded"
      onSubmit={[Function]}
    >
      <h2
        className="w-3/4 text-center text-xl text-bh-black mb-4"
      >
        Search on GitHub
      </h2>
      <input
        className="md:w-3/4 text-center text-bh-blue font-bold rounded px-1 py-2 mb-4"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <button
        className="w-1/2 flex items-center justify-center rounded-md bg-bh-black hover:bg-black text-white px-1 py-2 transition-all"
        type="submit"
      >
        Go!
      </button>
    </form>
  </section>
</div>
`;
